src/monosynth.js

```diff
@@ -0,0 +1,358 @@
+
+    // filter
+    this.filter = new p5.Filter('highpass');
+    this.filter.set(5, 1);
+
+    // oscillator --> env --> filter --> this.output (gain) --> p5.soundOut
+
+    this.oscillator.connect(this.filter);
+    this.env.setInput(this.oscillator);
+    this.env.connect(this.filter);
+    this.filter.connect(this.output);
+
+    this.oscillator.start();
+
+    //Audiovoices are connected to soundout by default
+
+    this._isOn = false;
+
+    p5sound.soundArray.push(this);
+  };
+
+
+  /**
+   *  Used internally by play() and triggerAttack()
+   *  to set the note of this.oscillator to a MIDI value.
+   *
+   *  Synth creators with more complex setups may have overridden
+   *  the oscillator chain with more than one oscillator,
+   *  and should override this method accordingly.
+   *  
+   *  @param   {Number} note           midi value of a note, where 
+   *                                   middle c is 60
+   *  @param   {Number} [secondsFromNow] (optional) a time (in seconds
+   *                                     from now) at which
+   *                                     to set the note
+   *  @private
+   */
+  p5.MonoSynth.prototype._setNote = function(note, secondsFromNow) {
+    var freqVal = p5.prototype.midiToFreq(note);
+    this.oscillator.freq(freqVal, 0, secondsFromNow);
+  };
+
+  /**
+     *  Play tells the MonoSynth to start playing a note
+     *  
+     *  @method playNote 
+     *  @param  {Number} [note] midi note to play (ranging from 0 to 127 - 60 being a middle C)
+     *  @param  {Number} [velocity] velocity of the note to play (ranging from 0 to 1)
+     *  @param  {Number} [secondsFromNow]  time from now (in seconds) at which to play
+     *  @param  {Number} [sustainTime] time to sustain before releasing the envelope
+     *  
+     */  
+
+  p5.MonoSynth.prototype.play = function (note, velocity, secondsFromNow, susTime) {
+    // set range of env (TO DO: allow this to be scheduled in advance)
+    var vel = velocity || 1;
+
+    this.triggerAttack(note,velocity,secondsFromNow);
+    this.triggerRelease(secondsFromNow + susTime);
+  };
+
+  /**
+     *  Trigger the Attack, and Decay portion of the Envelope.
+     *  Similar to holding down a key on a piano, but it will
+     *  hold the sustain level until you let go. 
+     *
+     *  @param  {Number} secondsFromNow time from now (in seconds)
+     *  @param  {Number} [velocity] velocity of the note to play (ranging from 0 to 1)
+     *  @param  {Number} [secondsFromNow]  time from now (in seconds) at which to play
+     *  @method  triggerAttack
+     */  
+  p5.MonoSynth.prototype.triggerAttack = function (note, velocity, secondsFromNow) {
+
+    var now = p5sound.audiocontext.currentTime;
+    var tFromNow = secondsFromNow || 0;
+    var t = now + tFromNow;
+    var n = p5.prototype.midiToFreq(note);
+    
+    this._isOn = true;
+    this.oscillator.freq(n, 0, t);
+    this.env.ramp(this.output, t, velocity);
```

Comment:
I think these values should be `secondsFromNow`, because they will add `audiocontext.currentTime` - so don't add `now` just yet